datasetinsights/io/download.py

```diff
@@ -67,6 +67,21 @@ def download_file(source_uri: str, dest_path: str, use_cache: bool = True):
     return dest_path
 
 
+def checksum_matches(filepath, expected_checksum, algorithm="CRC32"):
+    """ Check if the checksum matches
+
+    Args:
+        filepath (str): the doaloaded file path
+        expected_checksum (int): expected checksum of the file
+        algorithm (str): checksum algorithm. Defaults to CRC32
+
+    Returns:
+        True if the file checksum matches.
+    """
+    computed = compute_checksum(filepath, algorithm)
+    return computed == expected_checksum
+
+
 def validate_checksum(filepath, expected_checksum, algorithm="CRC32"):
```

Comment:
How will we now validate checksum since `DatasetDownloader` is not couple to dataset class. How would we know the type of dataset being downloaded?

Reply:
Why is the type of dataset being downloader relevant for what the checksum is? Shouldn't the checksum be stored together with the dataset instead of inside the repo?

Reply:
@adason  and I discussed about this. We decided to put the source checksums in the repo, which is easy to implement.

Reply:
Checksum is not part of this review. Currently for UnitySimulationDownloader there are no way to validate checksums as they are not provided in the manifest file.